packages/utils/upgrade/src/cli/index.ts

```diff
@@ -1,63 +1,38 @@
-import { Command, program } from 'commander';
-import chalk from 'chalk';
-import os from 'node:os';
+import { program, Option, InvalidOptionArgumentError } from 'commander';
+import assert from 'node:assert';
 
 import { version } from '../../package.json';
+import { isVersion, VersionRelease } from '../core';
 
-/**
- * Adds a command to the program and attach common options
- */
-const command = (name: string, description: string): Command => {
-  return program
-    .command(name)
-    .description(description)
-    .option('--dry-run', "Run the upgrade, but don't update the files", false)
-    .option('-d, --debug', 'Get more logs in debug mode', false)
-    .option('-s, --silent', "Don't log anything", false)
-    .option('-p, --project-path <project-path>', 'Path to the Strapi project');
-};
+import type { CLIOptions } from '../types';
 
-// Register commands
-
-command('next', 'Upgrade your Strapi application to the next major version').action(
-  async (options) => {
-    const { next } = await import('./commands/next.js');
-
-    return next(options);
-  }
-);
-
-command('latest', 'Upgrade your Strapi application to the latest version').action(
-  async (options) => {
-    const { latest } = await import('./commands/latest.js');
-
-    return latest(options);
-  }
-);
-
-command('fix-current', 'Run missing upgrades for the current major version').action(
-  async (path, options) => {
-    const { fixCurrent } = await import('./commands/fix-current.js');
-
-    return fixCurrent({ path, ...options });
-  }
-);
-
-// Miscellaneous
+const ALLOWED_TARGETS =
+  'Allowed choices are "<number>.<number>.<number>", major, minor, patch, latest.';
```

Comment:
```suggestion
  'Allowed choices are major, minor, patch, latest, or a specific version number in the form "x.x.x"';
```
Something like this might be clearer, I was confused at first

Reply:
Actually, if we switch to use semver parsing methods, we could just write something like "a valid semver version string"

Reply:
Thanks for the suggestion, will change the description

> Actually, if we switch to use semver parsing methods, we could just write something like "a valid semver version string"

But we don't want to allow any kind of semver string like semver with build or 4.x.x (although the later could be smth we implement later on). The issue with the "official" semver is that it allows way too much compared to our strategies + it's not typed well (or at all)